src/decider/agent/DeciderAgent.php

```diff
@@ -407,4 +435,33 @@ protected function areNomosMonkNinkaAgreed($licenseMatches)
     }
     return true;
   }
+
+  /**
+   * @brief Check if findings by all agents are same or not
+   * @param LicenseMatch[][] $licenseMatches
+   * @return boolean True if they match, false otherwise
+   */
+  protected function areNomosOrMonkAndOJOAgreed($licenseMatches)
+  {
+    $scanners = array('ojo','monk','nomos');
+    $currentFinding = array();
+    if (array_key_exists('ojo', $licenseMatches) === FALSE) {
+      return false;
+    }
+    foreach ($scanners as $scanner) {
+      if (array_key_exists($scanner, $licenseMatches) === FALSE) {
+        continue;
+      }
+      foreach ($licenseMatches[$scanner] as $licenseMatch) {
+        $licId = $licenseMatch->getLicenseId();
+        if (empty($currentFinding)) {
```

Comment:
Yes. Actually the first scanner finding that will be sent to $currentFinding will be OJO. So if all other scanner findings matches the existing finding then it will be a decision and if there is only OJO finding it will be a decision as well.

Reply:
For example, if the matches of `ojo` are the licenseIds `10` and `20`, then the logic would first set `$currentFinding` to `[10]` (since it was previously empty). And in the second run of the innermost loop (`$licId = 20`) it will see that `$currentFinding` is not empty and after that compare `in_array($licId, $currentFinding) === FALSE`. This can be rewritten as `in_array(20, [10]) === FALSE` and evaluates to `true` and thus return `false`. This means that your code here always returns `false`, if at least one scanner has more then one finding.

Reply:
yes correct.

Reply:
what about `SPDX-License-Identifier: {some license} AND {some other license}`?

In every case this is very implicit and hidden below some complicated logic. Why not make that explicit and use a single variable for the single possible license instead of a list used as an optional? You can even early exit, if more then one license was found by `ojo`.

Reply:
Then i hope OJO will not have any agent finding. so it should exit initially.

Reply:
This means, that `ojo` can't handle `SPDX-License-Identifier: {some license} AND {some other license}` at all?

Reply:
yes correct.

Reply:
But `{some license} OR {some other license}` would be `[{some license},{some other license},dual-license]`, and with `AND` just drop the `dual-license`?

I still think that this should be made explicit in the function, e.g. by something like:
```php
        $findingsByOjo = $this->getLicenseIdsOfMatchesForScanner('ojo', $licenseMatches);
        if (count($findingsByOjo) > 1) {
          return false; // We only conclude single licenses
        }
```